### PR TITLE
Move SyslogAppender into separate onprem bundle

### PR DESCRIPTION
**Pull request by cziegeler**
This implements #3401

### CODE REVIEW

File: bundle-onprem/pom.xml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.jackrabbit</groupId>
+                    <artifactId>jackrabbit-api</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <!-- End Test dependencies -->
+        <!-- the aem-sdk-api should be the last dependency, as other dependencies should take precedence in case of package clashes -->
+        <dependency>
```
Comment (kwin) [inline bundle-onprem/pom.xml:114]: This is the wrong dependency for AEM 6.5

File: bundle-onprem/pom.xml [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+            </plugin>
+        </plugins>
+    </build>
+
+    <dependencies>
+        <dependency>
```
Comment (kwin) [inline bundle-onprem/pom.xml:67]: For a new bundle we should IMHO start with the most recent dependencies (i.e. JUnit5)

File: bundle-onprem/pom.xml [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+            <artifactId>org.apache.sling.testing.sling-mock.junit4</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.wcm</groupId>
+            <artifactId>io.wcm.testing.aem-mock.junit4</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.sling</groupId>
```
Comment (kwin) [inline bundle-onprem/pom.xml:93]: Hopefully we don't need this legacy dep.

File: bundle-onprem/pom.xml (hunk 1 above)
Comment (cziegeler) [reply bundle-onprem/pom.xml:156]: good point, fixed

File: bundle-onprem/pom.xml (hunk 3 above)
Comment (cziegeler) [reply bundle-onprem/pom.xml:134]: Right now we do, I just ported the test as is

File: bundle-onprem/pom.xml [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    <!-- ====================================================================== -->
+
+    <artifactId>acs-aem-commons-bundle-onprem</artifactId>
+    <name>ACS AEM Commons Bundle - On Prem</name>
+    <description>Core ACS AEM Commons OSGi Bundle for AEM on prem only</description>
+   <build>
+        <plugins>
+            <plugin>
+                <groupId>biz.aQute.bnd</groupId>
+                <artifactId>bnd-maven-plugin</artifactId>
```
Comment (kwin) [inline bundle-onprem/pom.xml:42]: I am wondering whether we should make this a fragment of `com.adobe.acs.acs-aem-commons-bundle` similar to the Cloud bundle in order to consume non-exported classes from it.

Comment (cziegeler) [reply bundle-onprem/pom.xml:42]: I thought about this, too - not sure how likely that use case actually is and therefore decided to go with a plain old bundle.

File: bundle-onprem/pom.xml (hunk 3 above)
Comment (cziegeler) [reply bundle-onprem/pom.xml]: Updated the test to use Mockito for everything

File: bundle-onprem/pom.xml (hunk 2 above)
Comment (cziegeler) [reply bundle-onprem/pom.xml:69]: Updated to JUnit5

File: bundle-onprem/pom.xml (hunk 4 above)
Comment (kwin) [reply bundle-onprem/pom.xml:42]: ok, we can do it later on if needed.

Comment (cziegeler) [reply bundle-onprem/pom.xml:42]: @kwin  Thanks, is there anything else to be done?

**Comment by cziegeler** [comment]
@kwin Could you please merge this PR - or is there anything outstanding?

**Review by kwin — APPROVED** [review]

**Pull request merged**